spec/waypoint_spec.js

```diff
@@ -338,10 +338,20 @@ describe('<Waypoint>', function() {
   describe('when the scrollable parent does not have positioning', () => {
     beforeEach(() => {
       delete this.parentStyle.position;
+      this.props.onEnter.calls.reset();
+      this.props.onLeave.calls.reset();
+    });
+
+    it('does not call handlers to begin with', () => {
+      expect(this.subject).not.toThrow();
+      expect(this.props.onEnter).not.toHaveBeenCalled();
+      expect(this.props.onLeave).not.toHaveBeenCalled();
     });
 
-    it('throws an error', () => {
-      expect(this.subject).toThrow();
+    it('does call handlers when node becomes visible', () => {
+      this.parentStyle.position = 'relative';
+      expect(this.subject).not.toThrow();
+      expect(this.props.onEnter).toHaveBeenCalled();
     });
```

Comment:
In order for this to test the right thing, you need to do something like this:

```js
    it('does call handlers when node becomes visible', () => {
      this.component = this.subject();
      ReactDOM.findDOMNode(this.component).style.position = 'relative';
      scrollNodeTo(this.component, 10);
      expect(this.props.onEnter).toHaveBeenCalled();
    });
```

`this.subject()` will make the component mount on the page. The `this.parentStyle.position` only applies to the first render (inside `this.subject()`). So what you're really testing here isn't a transition from invisible to visible, it's just making sure that `onEnter` is called on initial render. My modification means we alter the DOM node after it has been rendered (which is what we want), and then the scroll makes sure that we recalculate positioning.